### DISCUSSION TITLE
Does signing of timecards exist?

### DISCUSSION THREAD

**Discussion by beaug777** [kimai/kimai#3662] [Questions & Answers]
I would love for my company to continue to use kimai2.  However, our customers are requiring us to use a timecard system where employees are able to "sign" their timecards at the end of a pay-period.
Such a feature would be convenient as well to know that everyone's hours are up-to-date before I run payroll.
Does this feature exist currently (possibly in a plugin)?  What is the likelyhood of such a feature being added?

**Comment by kevinpapst** [comment]
What does "sign their timecard" mean?

**Comment by beaug777** [comment]
The concept is, an employee can fill-out their timecard and make adjustments throughout the week (or two-weeks).  Before I can bill a customer for the employees hours I have to check with the employee and ask, "Is your timecard ready and accurate?"
Instead of asking each employee individually, the employee would click "sign timecard" to attest to the accuracy and completeness.  Their timecard for the pay period would then be "locked" from edits until they click "unsign".  However, at the time we "export" data for payroll or to create a customer's invoice, the "unlock" button would no longer be available to the employee.

**Reply by kevinpapst** [reply]
Thanks for the explanation. You can use the lockdown mode to disallow edits from a certain date on.

**Reply by beaug777** [reply]
Thanks for the quick responses!  Unfortunately, without an approval system and audit trail, I won't be able to be "DCAA compliant" while using this software.  Keep up the good work you've done here, I have been truly impressed!
